k8s/scripts/inspect.sh

```diff
@@ -0,0 +1,169 @@
+
+  log_info "Copy k8s diagnostics to the final report tarball"
+  k8s kubectl version &>"$INSPECT_DUMP/k8s-version.log"
+  k8s status &>"$INSPECT_DUMP/k8s-status.log"
+  k8s get &>"$INSPECT_DUMP/k8s-get.log"
+  k8s kubectl get cm k8sd-config -n kube-system -o yaml &>"$INSPECT_DUMP/k8s.k8sd/k8sd-configmap.log"
+  k8s kubectl get cm -n kube-system &>"$INSPECT_DUMP/k8s-configmaps.log"
+
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8s-dqlite/cluster.yaml "$INSPECT_DUMP/k8s.k8s-dqlite/k8s-dqlite-cluster.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8s-dqlite/info.yaml "$INSPECT_DUMP/k8s.k8s-dqlite/k8s-dqlite-info.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8sd/state/database/cluster.yaml "$INSPECT_DUMP/k8s.k8sd/k8sd-cluster.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8sd/state/database/info.yaml "$INSPECT_DUMP/k8s.k8sd/k8sd-info.yaml"
+
+  ls -la /var/snap/k8s/common/var/lib/k8s-dqlite &>"$INSPECT_DUMP/k8s.k8s-dqlite/k8s-dqlite-files.log"
+  ls -la /var/snap/k8s/common/var/lib/k8sd &>"$INSPECT_DUMP/k8s.k8sd/k8sd-files.log"
+}
+
+function collect_service_diagnostics {
+  local service
+  service=$1
+
+  mkdir -p "$INSPECT_DUMP/$service"
+
+  local status_file
+  status_file="$INSPECT_DUMP/$service/systemctl.log"
+
+  systemctl status "snap.$service" &>"$status_file"
+
+  local n_restarts
+  n_restarts=$(systemctl show "snap.$service" -p NRestarts | cut -d'=' -f2) 
+
+  printf -- "%s NRestarts=%s" "$service" "$n_restarts" >> "$INSPECT_DUMP/nrestarts.log"
+
+  if [ "$n_restarts" -gt 0 ]; then
+    log_warning "Service $service has restarted $n_restarts times due to errors"
+  fi
+
+  journalctl -n 100000 -u "snap.$service" &>"$INSPECT_DUMP/$service/journal.log"
+}
+
+function build_report_tarball {
+  local now_is
+  now_is=$(date +"%Y%m%d_%H%M%S")
+
+  tar -C "$(pwd)" -cf "$(pwd)/inspection-report-${now_is}.tar" inspection-report &>/dev/null
+  gzip "$(pwd)/inspection-report-${now_is}.tar"
+  log_success "Report tarball is at $(pwd)/inspection-report-$now_is.tar.gz"
+}
+
+if [ "$EUID" -ne 0 ]; then
+  printf -- "Elevated permissions are needed for this command. Please use sudo."
+  exit 1
+fi
+
+rm -rf "$INSPECT_DUMP"
+mkdir -p "$INSPECT_DUMP"
+
+printf -- 'Inspecting services\n'
+
+if is_control_plane_node; then
+  printf -- 'Running inspection on a control-plane node\n'
+  printf -- 'Inspection ran on a control plane node.' >"$INSPECT_DUMP/is-control-plane-node"
+else
+  printf -- 'Running inspection on a worker node\n'
+  printf -- 'Inspection ran on a worker node.' >"$INSPECT_DUMP/is-worker-node"
+fi
+
+control_plane_services=("k8s.containerd" "k8s.kube-proxy" "k8s.k8s-dqlite" "k8s.k8sd" "k8s.kube-apiserver" "k8s.kube-controller-manager" "k8s.kube-scheduler" "k8s.kubelet")
+worker_services=("k8s.containerd" "k8s.k8s-apiserver-proxy" "k8s.kubelet" "k8s.k8sd" "k8s.kube-proxy")
+
+if is_control_plane_node; then
```

Comment:
The content of this if statement is duplicated. Just move it into a function and invoke it with either the worker services or control_plane services depending